backend/mail-server/src/main/java/news/bombomemail/email/service/EmailService.java

```diff
@@ -0,0 +1,76 @@
+package news.bombomemail.email.service;
+
+import jakarta.mail.MessagingException;
+import jakarta.mail.Session;
+import jakarta.mail.internet.MimeMessage;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import news.bombomemail.article.service.ArticleService;
+import news.bombomemail.email.util.EmailContentExtractor;
+import org.springframework.dao.DataAccessException;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class EmailService {
+
+    private final Session mailSession;
+    private final ArticleService articleService;
+
+    public void processEmailFile(File emailFile) {
+        try (InputStream fileInputStream = new FileInputStream(emailFile)) {
+            MimeMessage mimeMessage = new MimeMessage(mailSession, fileInputStream);
+            final String contents = EmailContentExtractor.extractContents(mimeMessage);
+            boolean saved = articleService.save(mimeMessage, contents);
+            if (saved) {
+                deleteEmailFile(emailFile);
+                return;
+            }
+            moveToFailedDir(emailFile);
+        } catch (MessagingException | IOException e) {
+            log.error("이메일 파싱/입출력 오류: {}", emailFile.getName(), e);
+            moveToFailedDir(emailFile);
+        } catch (DataAccessException e) {
+            log.error("DB 오류: {}", emailFile.getName(), e);
+            moveToFailedDir(emailFile);
+        } catch (Exception e) {
+            log.error("예상치 못한 오류: {}", emailFile.getName(), e);
+            moveToFailedDir(emailFile);
+        }
+    }
```

Comment:
p3

아래처럼 `moveToFailedDir(emailFile)`을 공통으로 묶는건 어떨까요??


```suggestion
    public void processEmailFile(File emailFile) {
        try (InputStream fileInputStream = new FileInputStream(emailFile)) {
            MimeMessage mimeMessage = new MimeMessage(mailSession, fileInputStream);
            final String contents = EmailContentExtractor.extractContents(mimeMessage);
            boolean saved = articleService.save(mimeMessage, contents);
            if (saved) {
                deleteEmailFile(emailFile);
                return;
            }
        } catch (MessagingException | IOException e) {
            log.error("이메일 파싱/입출력 오류: {}", emailFile.getName(), e);
        } catch (DataAccessException e) {
            log.error("DB 오류: {}", emailFile.getName(), e);
        } catch (Exception e) {
            log.error("예상치 못한 오류: {}", emailFile.getName(), e);
        }
        moveToFailedDir(emailFile);
    }
```